functions.go

```diff
@@ -328,3 +328,49 @@ func errorStack(err error) []string {
 	}
 	return result
 }
+
+// CauseEquals tests, if this or any cause of this error is equal to another error, or it's cause
+//
+// For example:
+//		ErrNotFound := errors.New("Not found")
+//		inSome := errors.Trace(ErrNotFound)
+//		veryDeep := errors.Trace(inSome)
+//		returned := errors.Trace(veryDeep)
+//		if errors.CauseEquals(ErrNotFound, returned) {
+//			fmt.Println("Can return soft error 404")
+//		} else {
+//			fmt.Println("Serious error, returning with code 500")
+//		}
+func CauseEquals(err1, err2 error) bool {
```

Comment:
Most of external libraries (which does not use `juju/errors`) returns error form standard `"errors"` package. So somewhere I will still need to do casting.

I mean:
```go
type error interface {
	Error() string
}
```
does not have function `Cause`